geomstats/geometry/riemannian_metric.py

```diff
@@ -350,80 +350,113 @@ def mean(self, points,
 
         verbose: bool, optional
         """
+        if point_type != 'ball':
```

Comment:
How does your Frechet mean algorithm differ from the one implemented here? It seems that the stopping condition and the learning rate are the main differences, is there something else?

We should not distinguish the different cases of the Frechet mean computation by deciding by a condition on point_type. Instead:
- create another private method with your frechet mean algorithm
- call this specific method in your example
or:
- address issue #364 by creating a wrapper mean that decides which frechet mean algorithm to call based on an additional parameter called `method`, for example.

Reply:
> How does your Frechet mean algorithm differ from the one implemented here? It seems that the stopping condition and the learning rate are the main differences, is there something else?

There's something with the inner product that is not working on the coordinates type that I am using that I've still didn't work out. But you are right, with some additional work I should be able to include the frechet mean I use with the existing loop (still working on it). So for the moment I went with your proposed solution:

> address issue #364 by creating a wrapper mean that decides which frechet mean algorithm to call based on an additional parameter called method, for example.

Which I am not sure I did so well as a wrapper, I just added an extra parameter for the moment.